code/modules/mob/living/carbon/carbon.dm

```diff
@@ -51,41 +50,40 @@
 
 /mob/living/carbon/var/last_stomach_attack //defining this here because no one would look in carbon_defines for it
 
-/mob/living/carbon/relaymove(var/mob/user, direction)
-	if(user in src.stomach_contents)
-		if(last_stomach_attack + STOMACH_ATTACK_DELAY > world.time)	return
+/mob/living/carbon/relaymove(mob/user, direction)
+	if(LAZYLEN(stomach_contents))
+		if(user in stomach_contents)
+			if(last_stomach_attack + STOMACH_ATTACK_DELAY > world.time)
+				return
 
-		last_stomach_attack = world.time
-		for(var/mob/M in hearers(4, src))
-			if(M.client)
-				M.show_message(text("<span class='warning'>You hear something rumbling inside [src]'s stomach...</span>"), 2)
+			last_stomach_attack = world.time
+			for(var/mob/M in hearers(4, src))
+				if(M.client)
+					M.show_message(text("<span class='warning'>You hear something rumbling inside [src]'s stomach...</span>"), 2)
 
-		var/obj/item/I = user.get_active_hand()
-		if(I && I.force)
-			var/d = rand(round(I.force / 4), I.force)
+			var/obj/item/I = user.get_active_hand()
+			if(I && I.force)
+				var/d = rand(round(I.force / 4), I.force)
 
-			if(istype(src, /mob/living/carbon/human))
-				var/mob/living/carbon/human/H = src
-				var/obj/item/organ/external/organ = H.get_organ("chest")
-				if(istype(organ))
-					if(organ.receive_damage(d, 0))
-						H.UpdateDamageIcon()
+				if(istype(src, /mob/living/carbon/human))
+					var/mob/living/carbon/human/H = src
+					var/obj/item/organ/external/organ = H.get_organ("chest")
+					if(istype(organ))
+						if(organ.receive_damage(d, 0))
+							H.UpdateDamageIcon()
 
-				H.updatehealth("stomach attack")
+					H.updatehealth("stomach attack")
 
-			else
-				src.take_organ_damage(d)
+				else
+					take_organ_damage(d)
 
-			for(var/mob/M in viewers(user, null))
-				if(M.client)
-					M.show_message(text("<span class='warning'><B>[user] attacks [src]'s stomach wall with the [I.name]!</span>"), 2)
-			playsound(user.loc, 'sound/effects/attackblob.ogg', 50, 1)
+				for(var/mob/M in viewers(user, null))
+					if(M.client)
+						M.show_message(text("<span class='warning'><B>[user] attacks [src]'s stomach wall with the [I.name]!</span>"), 2)
+				playsound(user.loc, 'sound/effects/attackblob.ogg', 50, 1)
 
-			if(prob(src.getBruteLoss() - 50))
-				for(var/atom/movable/A in stomach_contents)
```

Comment:
this is unecessary; when a carbon of human gibs, they're forced out of `stomach_contents` anyway.